pkg/provider/aws/secretsmanager/fake/fake.go

```diff
@@ -22,15 +23,35 @@ import (
 
 // Client implements the aws secretsmanager interface.
 type Client struct {
-	valFn func(*awssm.GetSecretValueInput) (*awssm.GetSecretValueOutput, error)
+	ExecutionCounter int
+	valFn            map[string]func(*awssm.GetSecretValueInput) (*awssm.GetSecretValueOutput, error)
```

Comment:
In order to test the caching (miss out scenario), I had to accommodate for a map where the key is composed from the input key and version.